pkg/provider/device42/device42_api.go

```diff
@@ -0,0 +1,131 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package device42
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+)
+
+const (
+	DoRequestError         = "error: do request: %w"
+	errJSONSecretUnmarshal = "unable to unmarshal secret: %w"
+)
+
+type HTTPClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type API struct {
+	client   HTTPClient
+	baseURL  string
+	hostPort string
+	password string
+	username string
+	context  context.Context
```

Comment:
The problem with this context here is that it's alive / set as long as NewAPI's created instance lives.

It's never re-created. Which means that all the HTTP calls will share this single context. If it times out or has some kind of value or anything, it will never be re-done until a NewAPI is created/called.

Instead of doing that, just change `GetSecret` to accept a context as first parameter.

Either that, or when you do this:

```go
readSecretRequest, err := http.NewRequest("GET", endpointURL, http.NoBody)
```

Go with NewRequestWithContext instead and create a new context each time with timeout.

WDYT?

Reply:
@Skarlso thank you so much - yes that makes sense.
I updated the code to use `http.NewRequestWithContext` with a timeout of 30 seconds.  Now calls to the backed secret store will use their own context with timeout.
How does it look?

Reply:
That's better, so you can remove this context now as it's unused, right?

Reply:
I can't think of a reason why we'd need to use it there if we are creating a unique context there for each API call.  I just removed it, thanks again